src/main/java/harmonised/pmmo/mixin/CampfireMixin_Dev.java

```diff
@@ -0,0 +1,26 @@
+package harmonised.pmmo.mixin;
+
+import harmonised.pmmo.api.events.FurnaceBurnEvent;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.Container;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.CampfireBlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraftforge.common.MinecraftForge;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+@Mixin(CampfireBlockEntity.class)
+public class CampfireMixin_Dev {
```

Comment:
This is unnecessary.    And quite frankly i like that this dev-vs-production discrepancy exists.  It deters anyone who doesn't know modding from making knockoffs and publishing them on other sites.  As you discovered during your PR development, it's not hard to work around.  Therefore it serves as a deterrent and i want to keep that.

Reply:
You are totally right this is a deterrent. I'm just not sure if you think it through all the way :)
It deters both good actors and bad actors.

On the good actor's side - not every modder needs to have the patience and willpower to tinker with mixins and java's bytecode. There are many valuable contributors, especially the creative types, who will (understandably) refuse to touch this subject with a ten-foot pole. Hell - the only reason I am here is because I have a slightly bigger motivation than "I am bored and this looks like a cool project".

As for bad actors - what is your threat model here? Do you really believe that a person capable of creating a knockoff variant will not know how to remove a broken mixin?

I am not insisting on you merging this change, but I will keep it on my fork. Not out of spite, but out of pure pragmatism - I have better things than manual pre-prodbuild patching to spend my mental capacity on :)

Reply:
The issue was certainly more prevalent when 1.20 was the latest version.  There were multiple instances of users coming into our discord and asking for help with this mixin because they were trying to make a knockoff "just for their friends".  It was usually people who didn't bother to read the wiki to find out the mod already had the feature, or had a solution that didn't require modifying the source code.  

that said, since 1.21 does not have this mixin issue, I no longer have that shield, so perhaps your point about effectiveness of the deterrent is on point from a practicality standpoint.  I'll keep this in as it does make my life easier and prevents me from accidentally pushing a version with the local change.